packages/augur-ui/src/modules/market-cards/containers/migrate-market-notice.ts

```diff
@@ -16,48 +16,107 @@ import { DISMISSABLE_NOTICE_BUTTON_TYPES } from 'modules/reporting/common';
 import { DismissableNotice } from 'modules/reporting/common';
 import { selectReportingWinningsByMarket } from 'modules/positions/selectors/select-reporting-winnings-by-market';
 import { MarketReportClaimableContracts } from 'modules/types';
+import { createBigNumber } from 'utils/create-big-number';
 
 const mapStateToProps = (state: AppState, ownProps) => {
+  const { universe, blockchain } = state;
+  const { forkingInfo } = universe;
+  const isForking = forkingInfo !== null;
   const marketId = ownProps.marketId;
   const market = selectMarket(marketId);
-  const { endTime, reportingState } = market;
+  const { reportingState, endTime } = market;
 
-  const hasPassed = dateHasPassed(
+  let show = isForking;
+  let canMigrateMarkets = false;
+  let hasReleaseRepOnThisMarket = false;
+
+  const hasMarketEnded = dateHasPassed(
     state.blockchain.currentAugurTimestamp * 1000,
     endTime
   );
+
+  const hasForkPassed =
+    isForking &&
+    dateHasPassed(
+      blockchain.currentAugurTimestamp * 1000,
+      forkingInfo.forkEndTime
+    );
+
+  if (isForking && forkingInfo.winningChildUniverseId) {
+    const winning = universe.children.find(
+      c => c.id === forkingInfo.winningChildUniverseId
+    );
+    if (createBigNumber(winning.usersRep || ZERO).gt(ZERO)) {
+      canMigrateMarkets = true;
+    }
+  }
+
+  const marketNeedsMigrating =
+    hasForkPassed &&
+    reportingState !== REPORTING_STATE.FINALIZED &&
+    reportingState !== REPORTING_STATE.AWAITING_FINALIZATION;
+
   const releasableRep = selectReportingWinningsByMarket(state);
   let hasReleaseRep = releasableRep.totalUnclaimedRep.gt(ZERO);
-  if (hasReleaseRep && releasableRep.claimableMarkets && releasableRep.claimableMarkets.unclaimedRep) {
-    hasReleaseRep = releasableRep.claimableMarkets.marketContracts.filter(c => c.marketId === marketId).length > 0;
-  }
-  const show =
-    !!(
-      state.universe.forkingInfo &&
-      (state.universe.forkingInfo.winningChildUniverseId || hasReleaseRep)
-    ) &&
-    (reportingState !== REPORTING_STATE.FINALIZED &&
-      reportingState !== REPORTING_STATE.AWAITING_FINALIZATION);
 
+  if (
+    hasReleaseRep &&
+    releasableRep.claimableMarkets &&
+    releasableRep.claimableMarkets.unclaimedRep
```

Comment:
Shouldn't this be testing `releasableRep.claimableMarkets.marketContracts` instead of `releasableRep.claimableMarkets.unclaimedRep`?

Reply:
only need to know if the user has unclaimedRep. The getter does have contracts in collection that the user has already claimed so the unclaimedRep would be "0". we don't want those.